### PR TITLE
Updatign to Android O and updating FCM libraries

### PR DESCRIPTION
**Pull request by secondsun**

### CODE REVIEW

File: gradle.properties [hunk 1]
```diff
@@ -1,7 +1,7 @@
 # Android Stuff
 
-VERSION_CODE=410
-VERSION_NAME=4.1.0
+VERSION_CODE=422
```
Comment (danielpassos) [inline gradle.properties]: Why 422 instead of 420? Anyway it's only a number :grimacing:

File: library/build.gradle [hunk 2]
```diff
@@ -1,12 +1,12 @@
 apply plugin: 'com.android.library'
 
 android {
-    compileSdkVersion 25
+    compileSdkVersion 26
     buildToolsVersion "25.0.3"
```
Comment (danielpassos) [inline library/build.gradle]: s/25.0.3/26.0.1?

File: library/build.gradle [hunk 3]
```diff
@@ -23,9 +23,9 @@ android {
 }
 
 dependencies {
-    compile 'com.android.support:appcompat-v7:25.3.1'
+    compile 'com.android.support:appcompat-v7:26.0.+'
```
Comment (danielpassos) [inline library/build.gradle]: Just wondering if we should avoid use `+`

File: library/build.gradle [hunk 4]
```diff
@@ -23,9 +23,9 @@ android {
...
     compile 'org.jboss.aerogear:aerogear-android-pipe:3.1.0'
-    compile ('com.google.firebase:firebase-messaging:9.8.0', {
+    compile ('com.google.firebase:firebase-messaging:11.0.1', {
```
Comment (danielpassos) [inline library/build.gradle]: Have you tested with the last one `11.0.4`?

**Comment by damienomurchu** [comment]
Build failing due to license;

FAILURE: Build failed with an exception.

* What went wrong:
A problem occurred configuring project ':library'.
> You have not accepted the license agreements of the following SDK components:
  [Android SDK Platform 26].
  Before building your project, you need to accept the license agreements and complete the installation of the missing components using the Android Studio SDK Manager.
  Alternatively, to learn how to transfer the license agreements from one workstation to another, go to http://d.android.com/r/studio-ui/export-licenses.html

**Comment by danielpassos** [comment]
@damienomurchu it happening because you probably don't have the SDK 26, platform-tools or something like that and the Gradle is auto installing it without accepting the license

See: https://developer.android.com/studio/intro/update.html#download-with-gradle

**Comment by damienomurchu** [comment]
@danielpassos yeah, that's just the CircleCI failing (my local build is fine) - was this the issue we were having with Travis also?

**Comment by danielpassos** [comment]
@damienomurchu No in travis-ci we had problem with start the emulator. If there is a problem in circle-ci, problem we are not set it correctly => https://github.com/aerogear/aerogear-android-push/blob/master/circle.yml#L4-L9

File: library/build.gradle [hunk 5]
```diff
@@ -1,12 +1,12 @@
 apply plugin: 'com.android.library'
 
 android {
-    compileSdkVersion 25
+    compileSdkVersion 26
```
Comment (damienomurchu) [inline library/build.gradle]: circle.yml file will also need to be modified

**Comment by danielpassos** [comment]
@secondsun Can you also update the circle-ci? :kissing_heart:

**Comment by secondsun** [comment]
@danielpassos  Done!

**Comment by danielpassos** [comment]
@secondsun Build still failing because you forgot to update the `buildToolsVersion` see my comments above :wink:

**Pull request merged**